elephant/spike_train_generation.py

```diff
@@ -746,8 +805,7 @@ def single_interaction_process(
         if rate < 0 * pq.Hz:
             raise ValueError(
                 'rate (=%s) must be non-negative.' % str(rate))
-        rates_b = np.array(
-            [rate.magnitude for _ in range(n)]) * rate.units
+        rates_b = np.array([rate.magnitude] * n) * rate.units
```

Comment:
use np.repeat

Reply:
I tried, but then some strange error occurring. In the following for loops. Thus I capitulated.